cmd/osbuild-tests/main_test.go

```diff
@@ -161,6 +162,45 @@ gpgkey_urls = ["https://url/path/to/gpg-key"]
 	runComposer(t, "sources", "delete", "osbuild-test-addon-source")
 }
 
+// When gsl with version * was specified in the blueprint,
+// composer depsolved both x86_64 and i686 version of gsl.
+// This test case should prevent this from happening.
+// gsl is used because it has x86_64 and i686 versions on both RHEL and Fedora.
+// Also, gsl-devel package exists, which is not dependant on gsl and shouldn't
+// be depsolved.
+func TestWildcardDepsolve(t *testing.T) {
+	versionStrings := []string{"*", "2.*", ""}
+	for _, versionString := range versionStrings {
+		t.Run(versionString, func(t *testing.T) {
+			pushBlueprint(t, &blueprint.Blueprint{
+				Name:    "multilib",
+				Version: "0.0.1",
+				Packages: []blueprint.Package{
+					{
+						Name:    "gsl",
+						Version: versionString,
+					},
+				},
+			})
+
+			output := string(runComposer(t, "blueprints", "depsolve", "multilib"))
+			gslCount := 0
+			lines := strings.Split(output, "\n")
+			for _, line := range lines {
+				trimmedLine := strings.Trim(line, " ")
+				if strings.HasPrefix(trimmedLine, "gsl-") {
+					gslCount += 1
+				}
+			}
+
+			if !assert.Equalf(t, 1, gslCount, "gsl is specified %d-times in the depsolve, should be there only once", gslCount) {
+				t.Logf("depsolve output:\n%s", output)
+				t.FailNow()
+			}
+		})
+	}
+}
```

Comment:
IMO this kind of test is more suitable for blueprints_test.go, for example a slight modification to TestBlueprintDepsolveV0 will do the job. osbuild-tests should be for sanity testing of the cli commands with minimum assertions.